File: go/soapclient/soapClient.go
package soapclient

import (
	"bytes"
	"fmt"
	"io/ioutil"
	"net/http"
	"sync"
	"time"
)

var testCounts = []int{100, 200, 300}

const baseURL = "http://localhost:5001/soap"

var soapActions = map[string]string{
	"usuarios":  `<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:mus="http://musica.app/"><soapenv:Header/><soapenv:Body><mus:UsuariosRequest/></soapenv:Body></soapenv:Envelope>`,
	"musicas":   `<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:mus="http://musica.app/"><soapenv:Header/><soapenv:Body><mus:MusicasRequest/></soapenv:Body></soapenv:Envelope>`,
	"playlists": `<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:mus="http://musica.app/"><soapenv:Header/><soapenv:Body><mus:PlaylistsRequest/></soapenv:Body></soapenv:Envelope>`,
}

func testLoad(action, name string, numTest int) {
	start := time.Now()

	var wg sync.WaitGroup

	for i := 0; i < numTest; i++ {
		wg.Add(1)
		func(i int) {
			defer wg.Done()
			req, err := http.NewRequest("POST", baseURL, bytes.NewBuffer([]byte(action)))
			if err != nil {
				fmt.Printf("Request %d failed: %v\n", i, err)
				return
			}
			req.Header.Set("Content-Type", "text/xml")
			client := &http.Client{}
			resp, err := client.Do(req)
			if err != nil {
				fmt.Printf("Request %d failed: %v\n", i, err)
				return
			}
			defer resp.Body.Close()
			_, err = ioutil.ReadAll(resp.Body)
			if err != nil {
				fmt.Printf("Request %d failed: %v\n", i, err)
				return
			}
		}(i)
	}
	wg.Wait()

	end := time.Now()
	fmt.Printf("%s load test with %d requests completed in %v seconds\n", name, numTest, end.Sub(start).Seconds())
}

func RunTests() {
	for _, numTest := range testCounts {
		fmt.Printf("Running tests with %d requests:\n", numTest)
		testLoad(soapActions["usuarios"], "Usuarios", numTest)
		testLoad(soapActions["musicas"], "Musicas", numTest)
		testLoad(soapActions["playlists"], "Playlists", numTest)
	}
}

File: python/clients/soap_client.py
import requests
import time

BASE_URL = "http://localhost:5001/soap"
SOAP_ACTIONS = {
    "Usuarios": """<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:spy="spyne.examples.flask">
                     <soapenv:Header/>
                     <soapenv:Body>
                       <spy:ler_usuarios/>
                     </soapenv:Body>
                   </soapenv:Envelope>""",
    "Musicas": """<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:spy="spyne.examples.flask">
                    <soapenv:Header/>
                    <soapenv:Body>
                      <spy:ler_musicas/>
                    </soapenv:Body>
                  </soapenv:Envelope>""",
    "Playlists": """<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:spy="spyne.examples.flask">
                      <soapenv:Header/>
                      <soapenv:Body>
                        <spy:ler_playlists/>
                      </soapenv:Body>
                    </soapenv:Envelope>""",
}
TEST_COUNTS = [100, 200, 300]


def test_load(action, name, num_test):
    start = time.time()

    for i in range(num_test):
        try:
            response = requests.post(
                BASE_URL, data=action, headers={"Content-Type": "text/xml"}
            )
            response.raise_for_status()
        except requests.RequestException as e:
            print(f"Request {i} failed: {e}")
            if response is not None:
                print(f"Response content: {response.content}")

    end = time.time()
    print(
        f"{name} load test with {num_test} requests completed in {end - start:.6f} seconds"
    )


def run_tests():
    for num_test in TEST_COUNTS:
        print(f"Running tests with {num_test} requests:")
        for name, action in SOAP_ACTIONS.items():
            test_load(action, name, num_test)


if __name__ == "__main__":
    run_tests()
